core/src/env_platform/env_service.bal

```diff
@@ -61,6 +61,23 @@ service envservice on ep0 {
     }
     resource function postApplication(http:Caller caller, http:Request req, TreeRemovalForm body) returns error? {
 
+        http:Response response = new;
+        string authHeader = req.getHeader("Authorization");
+        [string, string]|error userInfoFromJWT = getUserInfoFromJWT(authHeader);
+        if (userInfoFromJWT is [string, string]) {
+            [string, string] [userId, userType] = userInfoFromJWT;
+            [boolean, string]|error saveApplicationResult = saveApplication(body, userId);
+            if (saveApplicationResult is error) {
+                response.statusCode = http:STATUS_INTERNAL_SERVER_ERROR;
```

Comment:
Shall we append a payload containing the error message like **"Internal Server Error occurred."** for this?

Reply:
Yeah I think that's a good idea.

Reply:
Yeah I will resolve all the requested changes by tomorrow.